portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,13 +27,30 @@
 public class DataServlet extends HttpServlet {
   private final List <String> messages = new ArrayList<>();
   private final Gson gson = new Gson();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    messages.add("Something crazy");
-    messages.add("Something simple");
-    messages.add("Something amazing");
     String json = gson.toJson(messages);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String username = getParameter(request , "username", "Anonymous");
```

Comment:
I was at first confused why "Anonymous" and "" were hardcoded, until I looked a #getParameter. Another style suggestion would be for hardcoded values to add an inline comment with the param name, to improve readability. 

ie.  getParameter(request , /* name=  * / "username", /* defaultValue=  */ "Anonymous");